tests/zero_code_change/test_training_with_no_grad_updates.py

```diff
@@ -0,0 +1,113 @@
+# Standard Library
+import os
+from tempfile import TemporaryDirectory
+
+# Third Party
+import torch
+from torch.utils.data import DataLoader, IterableDataset
+from utils import build_json
+
+# First Party
+from smdebug.trials import create_trial
+
+
+class TestDataset(IterableDataset):
+    """Test dataset to generate random samples."""
+
+    def __init__(self, num_items, num_samples):
+        self.num_items = num_items
+        self.num_samples = num_samples
+
+    def __iter__(self):
+        def gen_rand(num_samples):
+            num = 0
+            while num < num_samples:
+                yield torch.randint(self.num_items, size=(1,)).item()
+                num += 1
+
+        return gen_rand(self.num_samples)
+
+
+class TestModel(torch.nn.Module):
+    """Test model that tries to learn the mapping between index and embedding."""
+
+    def __init__(self, embedding_size):
+        super(TestModel, self).__init__()
+        self.fc = torch.nn.Sequential(
+            torch.nn.Linear(1, embedding_size, bias=True), torch.nn.Tanh()
+        )
+        self.fc.apply(init_weights("xavier"))
+
+    def forward(self, x):
+        return self.fc(x)
+
+
+def get_device():
+    return torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+def init_weights(method):
+    def _init_weights(m):
+        if type(m) == torch.nn.Linear:
+            if method == "xavier":
+                torch.nn.init.xavier_uniform_(m.weight)
+            elif method == "he":
+                torch.nn.init.kaiming_uniform_(m.weight)
+            else:
+                raise Exception("Unsupported initialization method: {}".format(method))
+
+    return _init_weights
+
+
+def do_training():
+    # Generate a matrix of fake random embeddings.
+    N = 1000
+    EMBEDDING_SIZE = 768
+    embedding_matrix = torch.nn.Embedding.from_pretrained(
+        torch.randn(N, EMBEDDING_SIZE, dtype=torch.float32)
+    )
+    embedding_matrix = embedding_matrix.to(get_device())
+
+    # Setup dataset + data loader.
+    BATCH_SIZE = 64
+    NUM_BATCHES = 100
+    train_dataset = TestDataset(num_items=N, num_samples=NUM_BATCHES * BATCH_SIZE)
+    train_loader = DataLoader(train_dataset, batch_size=BATCH_SIZE, shuffle=False)
+
+    # Setup model.
+    model = TestModel(embedding_size=EMBEDDING_SIZE).to(get_device())
+
```

Comment:
I think you need to set requires_grad to False, if the testcase is intended to check hook on parameters that do not require gradients. I am getting the following output:
```model = TestModel(10)
for parameter in model.parameters():
    print(parameter.requires_grad)
True
True
```